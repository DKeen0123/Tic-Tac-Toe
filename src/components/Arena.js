import React from 'react';

const Arena = props => {
  return (
    <div>
      {props.boxes.map(box => {
        return box;
      })}
    </div>
  );
};

export default Arena;
