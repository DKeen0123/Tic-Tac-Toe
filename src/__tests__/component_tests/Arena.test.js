import React from 'react';
import { shallow } from 'enzyme';
import Arena from '../../components/Arena';

describe('Arena', () => {
  const mockHandlePlayerMove = jest.fn();
  const mockBoxes = [<box key={1} />, <box key={2} />];
  const props = { handlePlayerMove: mockHandlePlayerMove, boxes: mockBoxes };
  let arena = shallow(<Arena {...props} />);

  it('renders correctly', () => {
    expect(arena).toMatchSnapshot();
  });

  it('loops through boxes props and renders them', () => {
    expect(arena.find('box').exists()).toBe(true);
  });
});
