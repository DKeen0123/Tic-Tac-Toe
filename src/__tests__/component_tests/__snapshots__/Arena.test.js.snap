// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arena renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Arena
    boxes={
        Array [
            <box />,
            <box />,
          ]
    }
    handlePlayerMove={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <box />,
        <box />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "box",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "box",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <box />,
          <box />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "box",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "box",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
